eks/cleanup.go

```diff
@@ -0,0 +1,110 @@
+package eks
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+
+	"github.com/gruntwork-io/kubergrunt/eksawshelper"
+	"github.com/gruntwork-io/kubergrunt/logging"
+)
+
+// CleanupSecurityGroup deletes the AWS EKS managed security group, which otherwise doesn't get cleaned up when
+// destroying the EKS cluster.
+// It also attempts to delete the security group left by ALB ingress controller.
+func CleanupSecurityGroup(
+	clusterArn string,
+	securityGroupID string,
+	vpcID string,
+) error {
+	logger := logging.GetProjectLogger()
+
+	// Get Region from ARN
+	region, err := eksawshelper.GetRegionFromArn(clusterArn)
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	// Get Cluster Name from ARN
+	clusterID, err := eksawshelper.GetClusterNameFromArn(clusterArn)
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	// Start new AWS session
+	sess, err := eksawshelper.NewAuthenticatedSession(region)
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	ec2Svc := ec2.New(sess)
+	logger.Infof("Successfully authenticated with AWS")
+
+	logger.Infof("Deleting security group %s", securityGroupID)
+	delSGInput := &ec2.DeleteSecurityGroupInput{
+		GroupId: aws.String(securityGroupID),
+	}
+	_, err = ec2Svc.DeleteSecurityGroup(delSGInput)
+	if err != nil {
+		// TODO?: check for Error.Code == 'InvalidGroup.NotFound'. This means it's already been deleted
```

Comment:
Yes we will want to do that, especially for SG IDs that are directly passed in.

Reply:
This was done in 2f91ed800dbcd2b6ec84bccf3c80644d1211a6da